### ISSUE TITLE
[Docs] Document how to use serve with Tuple observations

### ISSUE THREAD

**Issue by patopipp** [ray-project/ray#22022]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

There's no documentation that explains how to use Ray Rlib Serve with Tuple observation. The example provided uses the cartpole model that has an array as observation, but there are no examples with Tuple observations.

### Use case

We are training an RL model using ray. The model trains without any issues using PPOTrainer, but when we try to serve the policy using the [tutorial](https://docs.ray.io/en/latest/serve/tutorials/rllib.html#serve-rllib-tutorial) the JSON is empty and the response code is 500. I believe the issue is that we are not setting the correct observation. The observation space of the model is a Tuple, I cannot find any documentation that explains how to serve a model that has Tuple observation space.
This is the code we are using to serve the policy:

```
import gym
from starlette.requests import Request
import requests
from main import WordleEnv
import ray
import ray.rllib.agents.ppo as ppo
from ray import serve

@serve.deployment(route_prefix="/test")
class ServePPOModel:
    def __init__(self, checkpoint_path) -> None:
        self.trainer = ppo.PPOTrainer(
            env="my_env",
        )
        self.trainer.restore(checkpoint_path)

    async def __call__(self, request: Request):
        json_input = await request.json()
        obs = json_input["observation"]

        action = self.trainer.compute_action(obs)
        return {"action": int(action)}


if __name__ == '__main__':
    from ray import tune
    tune.register_env("my_env", lambda config: WordleEnv())

    serve.start()
    ServePPOModel.deploy("my_checkpoint")

    # That's it! Let's test it
    for _ in range(10):
        env = WordleEnv()
        obs = env.reset()

        print(f"-> Sending observation {obs}")
        obs_ = [o.tolist() for o in obs]
        print(obs_)

        resp = requests.get(
            "http://localhost:8000/test", json={"observation": obs_}
        )
        print(f"<- Received response {resp.json()}")
```

This is how the observation space looks:
```
self.observation_space = Tuple([
    MultiBinary(len(list(WordleEnv.letter_dict.values()))),
    MultiBinary(len(list(WordleEnv.letter_dict.values()))),
    MultiBinary(len(list(WordleEnv.letter_dict.values()))),
    MultiBinary(len(list(WordleEnv.letter_dict.values()))),
    MultiBinary(len(list(WordleEnv.letter_dict.values())))
])
```

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by maxpumperla** [comment]
@patopipp in the original tutorial `obs = env.reset()` is a numpy array, so we make it a list to then convert it all to json. Then on the server we de-serialise the json and read in `obs = json_input["observation"]`, which is now a list again. And in this case you're "lucky" because you can feed both plain lists and numpy arrays into your trainer's `compute_action`. (It's not necessary here, but the conceptually cleaner way would be to make `obs` a numpy array again).

If you have more complicated and potentially nested observations, `compute_action` does not know how to deal with e.g. the list of lists you created in `obs_ = [o.tolist() for o in obs]`. Concretely, you have to do this:

```python
obs = json_input["observation"]
# now convert obs into something that can be fed into your trainer.
```

Since you already know what your `trainer` expects, this shouldn't be a problem. I know this is a bit of a band-aid, and we'll definitely work on improving the docs. I just wanted to give you an idea how to approach this. This is the usual dance you have to do with HTTP, see for instance our [TensorFlow tutorial](https://docs.ray.io/en/latest/serve/tutorials/tensorflow.html), which needs both array creation and reshaping on the server.

**Comment by architkulkarni** [comment]
@maxpumperla Thanks for responding here!  Is this issue closed by #22030?
